### ISSUE TITLE
Proposals for Discord Configuration

### ISSUE THREAD

**Issue by StoneyJackson** [LibreFoodPantry/Community#63]
We need a Discord configuration that

* Maximizes transparency
* Minimizes spam capability
* Maximizes ease of on-boarding
* Maximizes safety of members
* Embodies this community's values and code of conduct

Discord has a lot of dials and knobs that we can use to tweak a policy to fit this community. We need to consider each option and propose an initial configuration, and ultimately implement that configuration.

**Comment by lpostner** [comment]
I feel like we have some information on how to set it up but I can't figure out where it is.  Was it on the closed issue about slack vs discord?  Can we link to it so I can ask if one of our students wants to work on this?

**Comment by kwurst** [comment]
@lpostner Yes, the information on how to turn off the random welcome messages and turn off the activity listing were on issue #17.

The random welcome messages are a server setting, so we never need to do it again (at least not for LFP because we already have the server and will just create new channels.) (https://github.com/LibreFoodPantry/Community/issues/17#issuecomment-518795882)

The activity feed is something that each user needs to do. (https://github.com/LibreFoodPantry/Community/issues/17#issuecomment-519239731)

**Comment by StoneyJackson** [comment]
More, the roles and permissions need to be considered and how new members will become members and ultimately be able to participate. Anyone (newcomers included) should be able to read any channel. Ideally, anyone can write to any channel, but we are concerned about spammers and adherence to code of conduct.

There are several options here depending on how likely we believe we'll get spammers. Stronger policies may require more work from us. For example, we could restrict newcomers to read/write #general and #read anything. And then have a "member" role that can read/write all channels. The challenge would be assigning the "member" role to each newcomer that we believe should have it. How do you decide this? If you need to on-board a bunch all at once, this could be tedious. Also, this creates an extra hoop for someone to jump through. Could a bot be added to a channel that would allow a newcomer to say the magic word and they would automatically be promoted? So again, we need someone to take a careful look at what's possible and then think about what should be done.

**Comment by kwurst** [comment]
Discord has some settings that restrict  privileges based on whether someone has a verified email address, and how long they have been on the channel.

Someone will have to take a closer look at how this works.

**Comment by StoneyJackson** [comment]
Maybe that's all we need? Length of time on channel?

**Comment by StoneyJackson** [comment]
Found this. Haven't yet read.

https://blog.discordapp.com/how-to-get-the-most-out-of-your-community-server-a049d126f2ac
